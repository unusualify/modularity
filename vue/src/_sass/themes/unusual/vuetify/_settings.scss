// themes/${theme}/vuetify/_settings.scss
@use '../abstract' as *;

@forward 'vuetify/settings' with (
  $table-row-height: 49px, // var(--v-table-row-height, 52px) !default
  $table-header-height: 49px, // 56px !default;
);
