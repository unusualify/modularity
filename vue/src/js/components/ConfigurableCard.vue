<template>
  <v-card class="ue-configurable-card" :class="[$bindAttributes().class, cardClass]">
    <!-- <template v-if="title" v-slot:title>
      <span class="font-weight-bold text-primary text-body-1">{{ title }}</span>
    </template> -->
    <slot name="title">
      <ue-title v-if="title" :text="title" :color="titleColor" :padding="`x-${titlePXNumber}`" class="pt-4"/>
    </slot>
    <div no-gutters class="ue-configurable-card__row"
      :style="rowStyle"
      :class="[
        $vuetify.display.smAndDown ? `ga-${mobileRowGap}` : '',
        rowMarginY ? `my-${rowMarginY}` : '',
        rowMarginX ? `mx-${rowMarginX}` : ''
      ]"
    >
      <div
        v-for="( segment,  segmentIndex) in itemsWithActions"
        :key="segmentIndex"
        :class="[
          'ue-configurable-card__col',
          !hideSeparator ? 'ue-configurable-card__col--seperator' : '',
          justifyCenterColumns ? 'ue-configurable-card__col--justify-center' : '',
          alignCenterColumns ? 'ue-configurable-card__col--align-center' : '',
          $isset(columnStyles[segmentIndex]) || colRatios.length > 0 ? `ue-configurable-card__col--unset-flex-basis` : '',
          columnClasses[segmentIndex] ?? '',
          colPaddingX ? `px-${colPaddingX}` : '',
          colPaddingY ? `py-${colPaddingY}` : ''
        ]"
        :style="getEffectiveColumnStyle(segmentIndex)"
      >
        <slot :name="`segment.${segmentIndex === '_actions' ? 'actions' : (parseInt(segmentIndex) + 1)}`"
          v-bind="{
            data: segment,
            actions: actions,
            actionProps: {actionIconMinHeight, actionIconSize}
          }"
        >
          <template v-if="actions.length && segmentIndex === '_actions'">
            <!-- <v-divider class="my-2"></v-divider> -->
            <div class="d-flex fill-height flex-wrap justify-space-evenly align-center">
              <v-btn
                v-for="(action, index) in actions"
                :key="index"
                class="mx-1 rounded"
                :min-width="actionIconMinHeight"
                :min-height="actionIconMinHeight"
                :size="actionIconSize"
                v-bind="action"
              />
            </div>
          </template>
          <template v-else-if="isObject(segment)">
            <div class="d-flex fill-height">
              <ue-property-list :data="segment" class="" noPadding/>
            </div>
          </template>
          <template v-else-if="isArray( segment)">
            <div class="d-flex fill-height">
              <ue-property-list :data="segment.map(item => [item])" class="" noPadding/>
            </div>

          </template>
          <template v-else>
            <div class="d-flex fill-height">
              <ue-dynamic-component-renderer :subject="segment"/>
            </div>
          </template>
        </slot>
      </div>
    </div>
  </v-card>
</template>

<script>
  export default {
    name: 'ue-configurable-card',
    props: {
      title: {
        type: String,
        default: ''
      },
      titleColor: {
        type: String,
      },
      titlePaddingX: {
        type: [Number, String],
      },
      titlePaddingY: {
        type: [Number, String],
      },
      items: {
        type: [Object, Array],
        required: true
      },
      actions: {
        type: Array,
        default: () => []
      },
      hideSeparator: {
        type: Boolean,
        default: false
      },
      maxSegments: {
        type: Number,
        default: null,
        validator: (value) => value === null || (value > 0 && value <= 12)
      },
      actionIconSize: {
        type: String,
        default: 'medium'
      },
      actionIconMinWidth: {
        type: Number,
        default: 44
      },
      actionIconMinHeight: {
        type: Number,
        default: 44
      },
      alignCenterColumns: {
        type: Boolean,
        default: false
      },
      justifyCenterColumns: {
        type: Boolean,
        default: false
      },
      rowMarginY: {
        type: [Number, String],
        default: 4
      },
      rowMarginX: {
        type: [Number, String],
      },
      colPaddingX: {
        type: [Number, String],
        default: 2
      },
      colPaddingY: {
        type: [Number, String],
      },
      columnStyles: {
        type: Object,
        default: () => ({}),
        // Example format: { 0: 'flex-basis: 50%', 1: 'flex-basis: 25%', 2: 'flex-basis: 25%' } or { 0: 'flex-grow: 2', 1: 'flex-grow: 1' }
      },
      columnClasses: {
        type: Object,
        default: () => ({}),
        // Example format: { 0: 'd-flex', '_actions': 'd-flex' }
      },
      colRatios: {
        type: Array,
        default: () => [],
        // Example format: [2, 1, 1] for 2:1:1 ratio or [3, 2, 1] for 3:2:1 ratio
      },
      rowMinHeight: {
        type: String,
        default: null
      },
      noActions: {
        type: Boolean,
        default: false
      },
      mobileRowGap: {
        type: [Number, String],
        default: 4
      }
    },
    computed: {
      titlePXNumber() {
        if (this.titlePaddingX) {
          return parseInt(this.titlePaddingX);
        }

        let padding = 0;

        if(this.rowMarginX) {
          padding += parseInt(this.rowMarginX);
        }

        if(this.colPaddingX) {
          padding += parseInt(this.colPaddingX);
        }

        return padding;
      },
      titlePYNumber() {
        return this.titlePaddingY ? parseInt(this.titlePaddingY) : 4;
      },
      cardClass() {
        return `ue-configurable-card--${this.effectiveSegmentCount}-columns`;
      },
      itemsWithActions() {
        // const lastKey = Object.keys(this.items).pop();
        const result = { ...this.items };
        if (this.actions.length && !this.noActions) {
          result['_actions'] = this.actions;
        }
        return result;
      },
      effectiveSegmentCount() {
        const itemCount = this.isArray(this.itemsWithActions)
          ? this.itemsWithActions.length
          : Object.keys(this.itemsWithActions).length;

        if (this.maxSegments === null) {
          return itemCount;
        }
        return Math.min(this.maxSegments, itemCount);
      },
      rowStyle() {
        return {
          ...(this.rowMinHeight ? { minHeight: this.rowMinHeight } : {})
        }
      },
      totalRatio() {
        // Calculate the total ratio to determine percentages
        if (!this.colRatios || this.colRatios.length === 0) {
          return this.effectiveSegmentCount;
        }

        return this.colRatios.reduce((sum, ratio, index) => {
          // Use the provided ratio or default to 1
          const value = ratio || 1;
          return sum + value;
        }, 0);
      }
    },
    methods: {
      isObject(value) {
        return __isObject(value);
      },
      isArray(value) {
        return __isArray(value);
      },
      getColumnStyleFromRatio(segmentIndex) {
        // Return null if no ratios are provided
        if (!this.colRatios || this.colRatios.length === 0) {
          return null;
        }

        // Convert segmentIndex to numeric index (handle '_actions' case)
        const colIndex = segmentIndex === '_actions' ? this.effectiveSegmentCount - 1 : parseInt(segmentIndex);

        // Use the provided ratio or default to 1
        const ratio = this.colRatios[colIndex] || 1;
        const percentage = (ratio / this.totalRatio) * 100;

        return {
          flex: `${ratio} 0 0`,
          maxWidth: `${percentage}%`
        };
      },
      getEffectiveColumnStyle(segmentIndex) {
        // Priority: 1. Custom columnStyles, 2. colRatios, 3. Default CSS classes
        const customStyle = this.columnStyles[segmentIndex];
        const ratioStyle = this.getColumnStyleFromRatio(segmentIndex);

        if (customStyle) {
          return customStyle;
        } else if (ratioStyle) {
          // Convert style object to CSS string
          return Object.entries(ratioStyle)
            .map(([key, value]) => `${key.replace(/([A-Z])/g, '-$1').toLowerCase()}: ${value}`)
            .join('; ');
        }

        return '';
      }
    }
  }
</script>

<style scoped lang="sass">
  .ue-configurable-card
    &__row
      display: flex
      flex-wrap: wrap
      max-width: 100%

      .ue-configurable-card__col
        &--justify-center:not(:first-child)
            > *:first-child
              justify-content: center

    .ue-configurable-card__col
      // flex: 1
      min-width: 0
      max-width: 100%

      &--seperator
        border-right: 1px solid rgba(0, 0, 0, 0.12)
      &--align-center
        > *:first-child
          align-items: center

      &:last-child
        border-right: none

    // Custom styles for different column counts
    @for $i from 1 through 12
      &--#{$i}-columns
        .ue-configurable-card__col:not([class*="ue-configurable-card__col--unset-flex-basis"])
          flex-basis: calc(100% / #{$i})
          max-width: calc(100% / #{$i})

    @media (max-width: 575px)
      &__row
        // flex-direction: column

      &__col
        flex-basis: 100% !important
        max-width: 100% !important
        padding: calc( 12 * $spacer/ 2) 0
        &--seperator
          border-right: none
          border-bottom: 1px solid rgba(0, 0, 0, 0.12)

        &:last-child
          border-bottom: none
</style>
