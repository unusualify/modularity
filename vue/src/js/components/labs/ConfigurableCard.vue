<template>
  <v-card class="ue-configurable-card" :class="[$bindAttributes().class, cardClass]">
    <!-- <template v-if="title" v-slot:title>
      <span class="font-weight-bold text-primary text-body-1">{{ title }}</span>
    </template> -->
    <ue-title :text="title" padding="x-4" class="pt-4"/>

    <div no-gutters class="ue-configurable-card__row">
      <div
        v-for="( segment,  segmentIndex) in itemsWithActions"
        :key="segmentIndex"
        :class="[
          'ue-configurable-card__col',
          !hideSeperator ? 'ue-configurable-card__col--seperator' : '',
          justifyCenterColumns ? 'ue-configurable-card__col--justify-center' : '',
          alignCenterColumns ? 'ue-configurable-card__col--align-center' : ''
        ]"
      >
        <template v-if="actions.length && segmentIndex === '_actions'">
          <!-- <v-divider class="my-2"></v-divider> -->
          <div class="d-flex fill-height flex-wrap justify-space-evenly align-center">
            <v-btn
              v-for="(action, index) in actions"
              :key="index"
              class="mx-1 rounded"
              :min-width="actionIconMinHeight"
              :min-height="actionIconMinHeight"
              :size="actionIconSize"
              v-bind="action"
            />
          </div>
        </template>
        <template v-else-if="isObject(segment)">
          <div class="d-flex fill-height">
            <PropertyList :data="segment" class="" noPadding/>
          </div>
        </template>
        <template v-else-if="isArray( segment)">
          <div class="d-flex fill-height">
            <PropertyList :data="segment.map(item => [item])" class="" noPadding/>
          </div>
          <!-- <v-list dense class="">
            <v-list-item v-for="(item, itemIndex) in segment" :key="itemIndex">
              {{ item }}
            </v-list-item>
          </v-list> -->
        </template>
        <template v-else>
          <div class="d-flex fill-height">
            <ue-dynamic-component-renderer :subject="segment"/>
          </div>
        </template>

      </div>
    </div>

    <!-- <v-card-actions v-if="actions && actions.length">
      <v-spacer></v-spacer>
      <v-btn v-for="(action, index) in actions" :key="index" :icon="action.icon" :color="action.color">
        <v-icon>{{ action.icon }}</v-icon>
      </v-btn>
    </v-card-actions> -->
  </v-card>
</template>

<script>
  import PropertyList from '@/components/labs/PropertyList.vue';

  export default {
    components: {
      PropertyList,
    },
    name: 'ue-configurable-card',
    props: {
      title: {
        type: String,
        default: ''
      },
      items: {
        type: [Object, Array],
        required: true
      },
      actions: {
        type: Array,
        default: () => []
      },
      hideSeperator: {
        type: Boolean,
        default: false
      },
      maxSegments: {
        type: Number,
        default: null,
        validator: (value) => value === null || (value > 0 && value <= 12)
      },
      actionIconSize: {
        type: String,
        default: 'medium'
      },
      actionIconMinWidth: {
        type: Number,
        default: 44
      },
      actionIconMinHeight: {
        type: Number,
        default: 44
      },
      alignCenterColumns: {
        type: Boolean,
        default: false
      },
      justifyCenterColumns: {
        type: Boolean,
        default: false
      }
    },
    computed: {
      cardClass() {
        return `ue-configurable-card--${this.effectiveSegmentCount}-columns`;
      },
      itemsWithActions() {
        // const lastKey = Object.keys(this.items).pop();
        const result = { ...this.items };
        if (this.actions.length) {
          result['_actions'] = this.actions;
          // __log(result)
        }
        return result;
      },
      effectiveSegmentCount() {
        const itemCount = this.isArray(this.itemsWithActions)
          ? this.itemsWithActions.length
          : Object.keys(this.itemsWithActions).length;

        if (this.maxSegments === null) {
          return itemCount;
        }
        return Math.min(this.maxSegments, itemCount);
      }
    },
    methods: {
      isObject(value) {
        return __isObject(value);
      },
      isArray(value) {
        return __isArray(value);
      },
    }
  }
</script>

<style scoped lang="sass">
  .ue-configurable-card
    &__row
      display: flex
      flex-wrap: wrap
      // padding-bottom: calc(12 * $spacer / 2)

      .ue-configurable-card__col
        &--justify-center:not(:first-child)
            > *:first-child
              justify-content: center

    .ue-configurable-card__col
      flex: 1
      min-width: 0
      padding: 0 calc($spacer * 4)
      margin: calc($spacer * 4) 0
      &--seperator
        border-right: 1px solid rgba(0, 0, 0, 0.12)
      &--align-center
        > *:first-child
          align-items: center

      &:last-child
        border-right: none

    // Custom styles for different column counts
    @for $i from 1 through 12
      &--#{$i}-columns
        .ue-configurable-card__col
          flex-basis: calc(100% / #{$i})
          max-width: calc(100% / #{$i})

    @media (max-width: 600px)
      &__row
        flex-direction: column

      &__col
        flex-basis: 100% !important
        max-width: 100% !important
        padding: calc( 12 * $spacer/ 2) 0
        &--seperator
          border-right: none
          border-bottom: 1px solid rgba(0, 0, 0, 0.12)

        &:last-child
          border-bottom: none
</style>
