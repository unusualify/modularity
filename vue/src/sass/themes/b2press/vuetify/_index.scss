// themes/${theme}/vuetify/_index.scss
@use '../abstract' as *;
@use 'vuetify' with (
  $color-pack: false,
  $body-font-family: $body-font-family,

  // $font-size-root: 2rem, // 1rem
  $line-height-root: $line-height-root, //1.5
  // $border-color-root: rgba(var(--v-border-color), var(--v-border-opacity)),
  // $border-radius-root: 4px,
  // $border-style-root: solid,

);
