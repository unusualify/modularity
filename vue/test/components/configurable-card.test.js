import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import ConfigurableCard from '../../src/js/components/labs/ConfigurableCard.vue'
import vuetify from '../../src/js/plugins/vuetify'
import UEConfig from '../../src/js/plugins/UEConfig'

describe('ConfigurableCard', () => {
  const createWrapper = (props = {}) => {
    return mount(ConfigurableCard, {
      props,
      global: {
        plugins: [UEConfig]
      }
    })
  }

  describe('Press Release Content', () => {
    it('renders simple array items', () => {
      const wrapper = createWrapper({
        title: 'Press Release Content',
        items: [
          'I have Press Release',
          ['Filename.docx', 'Filename.png'],
          {
            Date: '24.09.2024',
            Time: '15:00',
            'Time Zone': 'GMT +3'
          }
        ]
      })
      expect(wrapper.html()).toMatchSnapshot('press-release-content')
    })
  })

  describe('Package Details', () => {
    it('renders nested object structure', () => {
      const wrapper = createWrapper({
        title: 'Package Details',
        items: {
          details: {
            Package: 'Exclusive',
            Languages: 'English, German, French',
            'Distribution Countries': 'Germany, France, Türkiye'
          },
          content: {
            Content: 'Filename.docx',
            Media: 'Filename.png',
            Date: '24.09.2024',
            Time: '15:00',
            'Time Zone': 'GMT +3 (Türkiye)'
          },
          price: '$1,450 + VAT',
          status: 'Pending Payment'
        }
      })
      expect(wrapper.html()).toMatchSnapshot('package-details')
    })
  })

  describe('Addon Configuration', () => {
    it('renders info structure', () => {
      const wrapper = createWrapper({
        title: 'ADDON',
        items: {
          info: {
            title: 'Lorem Ipsum Dolor',
            description: 'At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis At vero eos et accusamus etiu.'
          }
        }
      })
      expect(wrapper.html()).toMatchSnapshot('addon-configuration')
    })
  })

  describe('Actions', () => {
    it('renders with action buttons', () => {
      const actions = [
        { icon: 'mdi-pencil', color: 'primary', onClick: () => {} },
        { icon: 'mdi-delete', color: 'error', onClick: () => {} },
        { text: 'complete', color: 'primary', class: 'px-6', onClick: () => {} }
      ]

      const wrapper = createWrapper({
        title: 'With Actions',
        items: { test: 'value' },
        actions
      })

      const actionButtons = wrapper.findAll('.v-btn')
      expect(actionButtons).toHaveLength(3)
      expect(wrapper.html()).toMatchSnapshot('with-actions')
    })

    it('handles action click events', async () => {
      const mockEdit = vi.fn()
      const mockDelete = vi.fn()

      const wrapper = createWrapper({
        title: 'Test Actions',
        items: { test: 'value' },
        actions: [
          { icon: 'mdi-pencil', color: 'primary', onClick: mockEdit },
          { icon: 'mdi-delete', color: 'error', onClick: mockDelete }
        ]
      })

      const buttons = wrapper.findAll('.v-btn')
      await buttons[0].trigger('click')
      expect(mockEdit).toHaveBeenCalled()

      await buttons[1].trigger('click')
      expect(mockDelete).toHaveBeenCalled()
    })
  })
})
