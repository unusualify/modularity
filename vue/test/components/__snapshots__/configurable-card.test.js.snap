// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfigurableCard > Actions > renders with action buttons > with-actions 1`] = `
"<div data-v-cd6087b7="" class="v-card v-theme--unusual v-card--density-default v-card--variant-elevated ue-configurable-card ue-configurable-card--2-columns">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--unusual v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <!-- <template v-if="title" v-slot:title>
      <span class="font-weight-bold text-primary text-body-1">{{ title }}</span>
    </template> -->
  <div data-v-e944dbcc="" data-v-cd6087b7="" class="text-body-1 text-primary text-uppercase font-weight-bold px-4 ma-0 text-left justify-start d-flex pt-4">With Actions
    <!-- <slot v-bind="{right}">

    </slot> -->
    <!-- <slot name="description" v-bind="{subTitle}">
      <div :class="[(!paddingReset ? paddingClasses: []), descriptionClasses]">
        {{ subTitle ?? '' }}
      </div>
    </slot> -->
  </div>
  <div data-v-cd6087b7="" no-gutters="" class="ue-configurable-card__row">
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">value</div>
    </div>
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <!-- <v-divider class="my-2"></v-divider> -->
      <div data-v-cd6087b7="" class="d-flex fill-height flex-wrap justify-space-evenly align-center"><button data-v-cd6087b7="" type="button" class="v-btn v-btn--elevated v-btn--icon v-theme--unusual bg-primary v-btn--density-compact v-btn--variant-elevated mx-1 rounded" style="min-height: 44px; min-width: 44px; text-transform: none;"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-pencil mdi v-icon notranslate v-theme--unusual v-icon--size-default" aria-hidden="true"></i></span>
          <!---->
          <!---->
        </button><button data-v-cd6087b7="" type="button" class="v-btn v-btn--elevated v-btn--icon v-theme--unusual bg-error v-btn--density-compact v-btn--variant-elevated mx-1 rounded" style="min-height: 44px; min-width: 44px; text-transform: none;"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-delete mdi v-icon notranslate v-theme--unusual v-icon--size-default" aria-hidden="true"></i></span>
          <!---->
          <!---->
        </button><button data-v-cd6087b7="" type="button" class="v-btn v-btn--elevated v-theme--unusual bg-primary v-btn--density-compact v-btn--variant-elevated mx-1 rounded px-6" style="min-height: 44px; min-width: 44px; text-transform: none;"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator="">complete</span>
          <!---->
          <!---->
        </button></div>
    </div>
  </div>
  <!-- <v-card-actions v-if="actions && actions.length">
      <v-spacer></v-spacer>
      <v-btn v-for="(action, index) in actions" :key="index" :icon="action.icon" :color="action.color">
        <v-icon>{{ action.icon }}</v-icon>
      </v-btn>
    </v-card-actions> -->
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`ConfigurableCard > Addon Configuration > renders info structure > addon-configuration 1`] = `
"<div data-v-cd6087b7="" class="v-card v-theme--unusual v-card--density-default v-card--variant-elevated ue-configurable-card ue-configurable-card--1-columns">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--unusual v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <!-- <template v-if="title" v-slot:title>
      <span class="font-weight-bold text-primary text-body-1">{{ title }}</span>
    </template> -->
  <div data-v-e944dbcc="" data-v-cd6087b7="" class="text-body-1 text-primary text-uppercase font-weight-bold px-4 ma-0 text-left justify-start d-flex pt-4">ADDON
    <!-- <slot v-bind="{right}">

    </slot> -->
    <!-- <slot name="description" v-bind="{subTitle}">
      <div :class="[(!paddingReset ? paddingClasses: []), descriptionClasses]">
        {{ subTitle ?? '' }}
      </div>
    </slot> -->
  </div>
  <div data-v-cd6087b7="" no-gutters="" class="ue-configurable-card__row">
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">
        <div data-v-2e888638="" data-v-cd6087b7="" class="ue-property-list">
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">title:</span><span data-v-2e888638="" class="text-caption">Lorem Ipsum Dolor</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">description:</span><span data-v-2e888638="" class="text-caption">At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis At vero eos et accusamus etiu.</span></div>
        </div>
      </div>
    </div>
  </div>
  <!-- <v-card-actions v-if="actions && actions.length">
      <v-spacer></v-spacer>
      <v-btn v-for="(action, index) in actions" :key="index" :icon="action.icon" :color="action.color">
        <v-icon>{{ action.icon }}</v-icon>
      </v-btn>
    </v-card-actions> -->
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`ConfigurableCard > Package Details > renders nested object structure > package-details 1`] = `
"<div data-v-cd6087b7="" class="v-card v-theme--unusual v-card--density-default v-card--variant-elevated ue-configurable-card ue-configurable-card--4-columns">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--unusual v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <!-- <template v-if="title" v-slot:title>
      <span class="font-weight-bold text-primary text-body-1">{{ title }}</span>
    </template> -->
  <div data-v-e944dbcc="" data-v-cd6087b7="" class="text-body-1 text-primary text-uppercase font-weight-bold px-4 ma-0 text-left justify-start d-flex pt-4">Package Details
    <!-- <slot v-bind="{right}">

    </slot> -->
    <!-- <slot name="description" v-bind="{subTitle}">
      <div :class="[(!paddingReset ? paddingClasses: []), descriptionClasses]">
        {{ subTitle ?? '' }}
      </div>
    </slot> -->
  </div>
  <div data-v-cd6087b7="" no-gutters="" class="ue-configurable-card__row">
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">
        <div data-v-2e888638="" data-v-cd6087b7="" class="ue-property-list">
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Package:</span><span data-v-2e888638="" class="text-caption">Exclusive</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Languages:</span><span data-v-2e888638="" class="text-caption">English, German, French</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Distribution Countries:</span><span data-v-2e888638="" class="text-caption">Germany, France, TÃ¼rkiye</span></div>
        </div>
      </div>
    </div>
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">
        <div data-v-2e888638="" data-v-cd6087b7="" class="ue-property-list">
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Content:</span><span data-v-2e888638="" class="text-caption">Filename.docx</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Media:</span><span data-v-2e888638="" class="text-caption">Filename.png</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Date:</span><span data-v-2e888638="" class="text-caption">24.09.2024</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Time:</span><span data-v-2e888638="" class="text-caption">15:00</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Time Zone:</span><span data-v-2e888638="" class="text-caption">GMT +3 (TÃ¼rkiye)</span></div>
        </div>
      </div>
    </div>
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">$1,450 + VAT</div>
    </div>
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">Pending Payment</div>
    </div>
  </div>
  <!-- <v-card-actions v-if="actions && actions.length">
      <v-spacer></v-spacer>
      <v-btn v-for="(action, index) in actions" :key="index" :icon="action.icon" :color="action.color">
        <v-icon>{{ action.icon }}</v-icon>
      </v-btn>
    </v-card-actions> -->
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`ConfigurableCard > Press Release Content > renders simple array items > press-release-content 1`] = `
"<div data-v-cd6087b7="" class="v-card v-theme--unusual v-card--density-default v-card--variant-elevated ue-configurable-card ue-configurable-card--3-columns">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--unusual v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <!-- <template v-if="title" v-slot:title>
      <span class="font-weight-bold text-primary text-body-1">{{ title }}</span>
    </template> -->
  <div data-v-e944dbcc="" data-v-cd6087b7="" class="text-body-1 text-primary text-uppercase font-weight-bold px-4 ma-0 text-left justify-start d-flex pt-4">Press Release Content
    <!-- <slot v-bind="{right}">

    </slot> -->
    <!-- <slot name="description" v-bind="{subTitle}">
      <div :class="[(!paddingReset ? paddingClasses: []), descriptionClasses]">
        {{ subTitle ?? '' }}
      </div>
    </slot> -->
  </div>
  <div data-v-cd6087b7="" no-gutters="" class="ue-configurable-card__row">
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">I have Press Release</div>
    </div>
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">
        <div data-v-2e888638="" data-v-cd6087b7="" class="ue-property-list">
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Filename.docx</span><span data-v-2e888638="" class="text-caption"></span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Filename.png</span><span data-v-2e888638="" class="text-caption"></span></div>
        </div>
      </div>
      <!-- <v-list dense class="">
            <v-list-item v-for="(item, itemIndex) in segment" :key="itemIndex">
              {{ item }}
            </v-list-item>
          </v-list> -->
    </div>
    <div data-v-cd6087b7="" class="ue-configurable-card__col ue-configurable-card__col--seperator">
      <div data-v-cd6087b7="" class="d-flex fill-height">
        <div data-v-2e888638="" data-v-cd6087b7="" class="ue-property-list">
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Date:</span><span data-v-2e888638="" class="text-caption">24.09.2024</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Time:</span><span data-v-2e888638="" class="text-caption">15:00</span></div>
          <div data-v-2e888638="" class="ue-property-list__item"><span data-v-2e888638="" class="text-caption font-weight-bold pr-1">Time Zone:</span><span data-v-2e888638="" class="text-caption">GMT +3</span></div>
        </div>
      </div>
    </div>
  </div>
  <!-- <v-card-actions v-if="actions && actions.length">
      <v-spacer></v-spacer>
      <v-btn v-for="(action, index) in actions" :key="index" :icon="action.icon" :color="action.color">
        <v-icon>{{ action.icon }}</v-icon>
      </v-btn>
    </v-card-actions> -->
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
